packages/edit-post/src/components/layout/index.js

```diff
@@ -71,6 +71,56 @@ const interfaceLabels = {
 	footer: __( 'Editor footer' ),
 };
 
+function useEditorStyles() {
+	const { hasThemeStyleSupport, editorSettings } = useSelect(
+		( select ) => ( {
+			hasThemeStyleSupport:
+				select( editPostStore ).isFeatureActive( 'themeStyles' ),
+			editorSettings: select( editorStore ).getEditorSettings(),
+		} )
+	);
+
+	// Compute the default styles.
+	const { defaultEditorStyles, presetStyles } = useMemo( () => {
+		const _presetStyles =
+			editorSettings.styles?.filter(
+				( style ) =>
+					style.__unstableType && style.__unstableType !== 'theme'
+			) ?? [];
+		return {
+			presetStyles: _presetStyles,
+			defaultEditorStyles: [
+				...editorSettings.defaultEditorStyles,
+				..._presetStyles,
+			],
+		};
+	}, [ editorSettings.defaultEditorStyles, editorSettings.styles ] );
+
+	// Has theme styles if the theme supports them and if some styles were not preset styles (in which case they're theme styles).
+	const hasThemeStyles =
+		hasThemeStyleSupport &&
+		presetStyles.length !== ( editorSettings.styles?.length ?? 0 );
+
+	// If theme styles are not present or displayed, ensure that
+	// base layout styles are still present in the editor.
+	if ( ! editorSettings.disableLayoutStyles && ! hasThemeStyles ) {
+		defaultEditorStyles.push( {
+			css: getLayoutStyles( {
+				style: {},
+				selector: 'body',
+				hasBlockGapSupport: false,
+				hasFallbackGapSupport: true,
+				fallbackGapValue: '0.5em',
+			} ),
+		} );
+	}
+
+	return useMemo(
```

Comment:
Do we need to memoize these values?

Reply:
Good point, likely not since the are already memoized or coming from the store

Reply:
Yep, it works without it 👍